decomposer_test.go

```diff
@@ -0,0 +1,80 @@
+package decimal
+
+import (
+	"testing"
+)
+
+func TestDecomposerRoundTrip(t *testing.T) {
+	list := []struct {
+		N string // Name.
+		S string // String value.
+		E bool   // Expect an error.
+	}{
+		{N: "Normal-1", S: "123.456"},
```

Comment:
I don't see any uses of an "assert" package, but, I added more test cases, it already checks with Decimal.Cmp and I added a check to the original S. Because this is a round trip test, I'm checking the final decimal against the original parsed number.